test/Microsoft.AspNet.Mvc.TagHelpers.Test/TagHelperOutputExtensionsTest.cs

```diff
@@ -80,6 +80,39 @@ public void CopyHtmlAttribute_DoesNotOverrideAttributes()
         }
 
         [Fact]
+        public void CopyHtmlAttribute_ThrowsWhenUnknownAttribute()
+        {
+            // Arrange
+            var invalidAttributeName = "hello2";
+            var tagHelperOutput = new TagHelperOutput(
+                "p",
+                attributes: new Dictionary<string, object>());
+            var tagHelperContext = new TagHelperContext(
+                allAttributes: new Dictionary<string, object>(StringComparer.Ordinal)
+                {
+                    { "hello", "world" }
+                },
+                items: new Dictionary<object, object>(),
+                uniqueId: "test",
+                getChildContentAsync: () =>
+                {
+                    var tagHelperContent = new DefaultTagHelperContent();
+                    tagHelperContent.Append("Something");
+                    return Task.FromResult<TagHelperContent>(tagHelperContent);
+                });
+
+            string expectedMessage = 
+                "The attribute 'hello2' does not exist in the TagHelperContext." + Environment.NewLine +
+                "Parameter name: attributeName";
+
+            // Act
+            var ex = Assert.Throws<ArgumentException>(() => { tagHelperOutput.CopyHtmlAttribute(invalidAttributeName, tagHelperContext); });
```

Comment:
we're not great about using the more-complete `Assert.Throws<ArgumentException>()` overloads.  but should switch pass the `paramName` argument i.e. `Assert.Throws<ArgumentException>("attributeName", () => ...)`

nit: bit messy. can remove the useless curlies: `() => tagHelperOutput.CopyHtmlAttribute(invalidAttributeName, tagHelperContext)`.